client/llb/exec.go

```diff
@@ -24,11 +24,7 @@ func NewExecOp(root Output, meta Meta, readOnly bool, md OpMetadata) *ExecOp {
 		readonly: readOnly,
 	}
 	e.mounts = append(e.mounts, rootMount)
-	if readOnly {
-		e.root = root
-	} else {
-		e.root = &output{vertex: e, getIndex: e.getMountIndexFn(rootMount)}
-	}
+	e.root = &output{vertex: e, getIndex: e.getMountIndexFn(rootMount)}
```

Comment:
Is this correct though? Shouldn't we handle it in `llbop/exec` instead? Skipping this output would make a sequence of execs with readonly rootfs to run in parallel. So in `llbop/exec` don't add it to the outputs and just release the mutableref as soon as the exec has completed.

Reply:
I was considering highlighting this particular portion for extra maintainer scrutiny since I@m really not sure ;-).

My thinking was that when readOnly was true this was actually being subverted further down so we wanted to treat it as writeable here, it ties in with the change just below at https://github.com/moby/buildkit/pull/324/files#diff-b6e1be4db0a6b8d1aada0afaaa76287dR155 which adds `m.target == pb.RootMount` to the output index stuff.

I'll take a look at the `llbop/exec` portions you suggest and see if I can make it work since that would keep the internals internal.